From 7841cc93050fa48a1fbd8847cec887f0360be9a3 Mon Sep 17 00:00:00 2001
From: Mark Washenberger <mark.washenberger@rackspace.com>
Date: Sun, 7 Oct 2012 02:12:52 +0000
Subject: [PATCH] FakeAuth not always admin

The FakeAuthMiddleware was defaulting to admin=True for request
contexts. This creates a situation where it is impossible to test
non-admin requests in some situations.

Change-Id: I949f708efed9f07a43b8506870c5b38fce4b3752
---
 glance/tests/unit/v1/test_api.py |    3 ++-
 glance/tests/utils.py            |    2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/glance/tests/unit/v1/test_api.py b/glance/tests/unit/v1/test_api.py
index dfb5e8a..de22657 100644
--- a/glance/tests/unit/v1/test_api.py
+++ b/glance/tests/unit/v1/test_api.py
@@ -103,7 +103,8 @@ class TestRegistryAPI(base.IsolatedUnitTest):
         """Establish a clean test environment"""
         super(TestRegistryAPI, self).setUp()
         self.mapper = routes.Mapper()
-        self.api = test_utils.FakeAuthMiddleware(rserver.API(self.mapper))
+        self.api = test_utils.FakeAuthMiddleware(rserver.API(self.mapper),
+                                                 is_admin=True)
         self.FIXTURES = [
             {'id': UUID1,
              'name': 'fake image #1',
diff --git a/glance/tests/utils.py b/glance/tests/utils.py
index cb9278d..92b6500 100644
--- a/glance/tests/utils.py
+++ b/glance/tests/utils.py
@@ -347,7 +347,7 @@ def minimal_add_command(port, name, suffix='', public=True):
 
 class FakeAuthMiddleware(wsgi.Middleware):
 
-    def __init__(self, app, is_admin=True):
+    def __init__(self, app, is_admin=False):
         super(FakeAuthMiddleware, self).__init__(app)
         self.is_admin = is_admin
 
